### ISSUE TITLE
Bad Debt Report in SAB

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#703]
In GitLab by @kdjstudios on Feb 19, 2018, 14:06

**Submitted by:** "Cheryl Hamelin" <cheryl.hamelin@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/4629914

**Server:** Internal

**Client/Site:** Atlanta

**Account:**  NA

**Issue:** 

When I try to pull up the bad debt report for Atlanta in SAB, there is no data. Please see screen shot below. When Cori pulls the same report, she gets data. (Also See below). I can get data for the Stockton site.

This is from mine.

![image001](/uploads/ba8cbb2874352cd6986131c5e3b01347/image001.jpg)

This is Cori’s for the same date.

![image002](/uploads/fb9cec76785f1857a54ede5035b4497b/image002.jpg)

**Comment by kdjstudios** [comment]
In GitLab by @pchaudhary on Feb 20, 2018, 06:36

@kdjstudios 

If you see closely the Billing Cycle Date on top of the report is different when it should be `01/27/2018`. Maybe April came soon for Cheryl. :smile:   

![Answernet](/uploads/de94e21530179593de604e63ba582081/Answernet.mp4)

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 20, 2018, 09:44

I sent an update to Cheryl, Thank you.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 22, 2018, 08:20

Cheryl wrote:

> If you go back to my original email to you, both reports are for the same date which is 1/27/18. One is from Cori's screen which shows the information and the blank one is from my screen. I just pulled the same report again using the 1/27/18 date and then for the 12/27/18 date. They are both blank. When you go to the billing cycle date field, there is not even a 1/21/18 listed.
> 
> Please recheck and have the issue corrected.
> 
> Thanks for your help.

My response:

> The attached video may show you how that could have happened to get different dates on the screenshots. But how you did that is beside the point.
> 
> I just want to confirm that you are now both showing matching reports, correct? If there is an issue could you please explain what the issue is now. Thanks and we look forward to your response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 22, 2018, 10:55

Cheryl response:


> I am confused by your statement that I got different dates on my screenshot.
> 
> I was on the phone with Cori and we were discussing the Bad Debt Report in SABilling. She requested that I pull up the Atlanta Bad Debt report for 1/27/18, which I did. The report I pulled does not have any information on it. (I sent you the screen shot). When Cori pulled up the report for the same date, hers showed information. She sent me a screen shot of what it looked like on her end. I sent that screenshot to you as well. Your email to me on 2/20/18 says the screen shots were for different dates. They were not. If you go back to the original email and look closely at the screen shots you will see they are for the same date. 
> 
> I later tried to pull the same report for the 12/27/18 date to see if there was information on it and there was not.
> 
> When I pulled the same report for Stockton, there was information.
> 
> The issue is, Cori can get information when she pulls the report and I cannot.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 22, 2018, 10:57

@smasih 

It appears the bad debt report is now not pulling any data for any sites. I tried to pull the report for Allentown, atlanta, and Bismark, all with 0 results, no matter what billing cycle I selected. Which seems highly irregular. Please resolve the issue.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 23, 2018, 05:44

@kdjstudios 

There is no issue in the Bad Debt report. 

Report plucks invoices if there is a `Bad Debt Code` present in their line items.

Here is the list of `Bad Debt` codes used in **Bad Debt Report**

* # Allentown   debt_code = "778"
* # Fair Oaks   debt_code = "7778"
* # Houston     debt_code = "7882"
* # Jonesboro   debt_code = "7778"
* # Miami       debt_code = "7778"
* # Orlando     debt_code = "7778"
* For all other sites **debt_code = "7778"**

So, either site debt code is not matching with `bad debt report codes` or there was no Invoices with `bad debt line item` for that billing cycle.

In Allen town, Bad debt code as per **bad debt report** was supposed to be `778` but was `7778`

![image](/uploads/4347dec7664823a7b069156cfec4b894/image.png)

In short Bad Debt Report is working we just need to change code from 7778 to 778 in Allentown and Atlanta had no invoices with code `7778` in their line items.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Feb 23, 2018, 07:27

@kdjstudios 

Whereas switching of site is concerned when we get accounts in bad debt report is because:

Allentown had `7778` as a bad debt code but system was expecting `778` so we had no result on the report.

As soon as user selects Atlanta which has `7778` code system re-runs the report  using code `7778` which it got from Atlanta and we get the result.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 23, 2018, 08:39

Update from Cheryl:

> Based on the little tutorial you sent me, in order to see what Cori got, I have to put in Allentown and pull it up and then go to Atlanta? Why would it be set up like that? Why doesn't Atlanta pull from the date that shows in the Billing Cycle field next to the Go button? It is strange to me that you have to go into a different site, have a different date and then change it to the site you want. 
> 
> I decided to investigate further so I looked at the report that Cori pulled up and if you look at the accounts on the report, they are Allentown accounts, not Atlanta. Allentown is site 061 and Atlanta is 053. I then looked up the accounts on the report and they definitely are part of Allentown, so even though you change the site name like the tutorial shows, you are still getting the information for the original site.
> 
> Please call me when you have a minute so we can get this figured out.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Feb 23, 2018, 08:41

My response:

> I do apologize as I will be in meetings and training throughout the rest of today. Just to clarify again; what the support team first pointed out was that the screenshots were showing two different reports. How that happened could be one many different methods, they simply provided the most common one. Please disregard all of that communication as it is confusing you and is not the true issue.  
> 
> The true issue: The support team looked further into this they found a discrepancy on the codes being used by the Allentown site. Simply, all other sites are using debt_code = "7778"; while Allentown is using debt_code = "778". So to solve your issue with the Bad debt report you may simply update the code to the correct value. This can be done at the site edit page.

**Comment by kdjstudios** [comment]
In GitLab by @smasih on Mar 8, 2018, 06:11

@kdjstudios 

I think the query was resolved so can we close this.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Mar 8, 2018, 08:48

@smasih @pchaudhary 

Agreed, We have not had any updates, marking for close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Apr 11, 2018, 13:16

closed
